### ISSUE TITLE
Revamped UI 

### ISSUE THREAD

**Issue by ImGunter** [mumble-voip/mumble#3608]
I think a revamped ui with much nicer and simpler menus would go a long way for mumble maybe adding image support could also help it stand out against the growing competition possibly some type of chat system using the open source signal protocol there’s a lot of things that can be done to make mumble an even more fantastic program.

**Comment by Kissaki** [comment]
Have you used the new UI themes in/from 1.3.0 yet?

What kind of simplifications would you like to see?

**Comment by ImGunter** [comment]
Yes I have tried it does look cleaner but I think it would be nice to have a cleaner ui similar to how ts5 looks just to make it more accessible to people I feel like it could thrive if it had that

**Comment by azlux** [comment]
Something like that talk to you ?
A friend have make mumble like discord (with gravit designer software).
![full with tchat](https://user-images.githubusercontent.com/11210117/58584593-df7fbe00-8256-11e9-8904-f4a6ff32f612.png)

**Comment by azlux** [comment]
@ImGunter for the image support, i've imagine some way to implement it #3597 .

**Comment by toby63** [comment]
I want to add my concept:
I am no artist, so it is not very clean and its also a bit fuzzy, but I think you can see what its about.

The basic idea is to have two seperate tabs.
This has mainly two reasons:
1. Three rows (like e.g. Discord) are a bit to much imo
2. I like the focus on one channel (as thats my usual way to spent my time on a server)

So first we got the **Server tab**:
- The general idea is to make icons and text bigger (I would maybe also make the users bigger than they are in the picture) and to give channels more details (symbols possible, short description and expandable) as well as a better visible seperation (with the boxes around them).
- The users also have more details (symbols, status (like afk), little versions of profile pictures) and direct functions (If you click one them, you can mute them, use volume control and get more information).
The green microfone presents the audio status (this would mean "talking").
- I also added a field for server news and announcements that can be closed by the user.
- On the right side more text can be added about the server and the channels.

![server_tab](https://user-images.githubusercontent.com/61703250/77207878-c94f3100-6afa-11ea-8e44-e86ab8012b56.png)

Second we got the **Channel tab**:
As you can see we have the chat on the left and a user list on the right.
The user profiles are expandable as well (if you click on the profile); see below.
For better visibility the messages are in boxes and the user names are coloured.
Improvements: Maybe I will add the Channel information as an expandable box above the userlist.

![channel_tab](https://user-images.githubusercontent.com/61703250/77208584-678fc680-6afc-11ea-9aa8-8d53c99c0763.png)

Then we got a **user profile in expanded view**:
The status (like afk) should be visible in the normal view as well.
Just like in the server tab you can mute them, use volume control and get more information directly by clicking on the buttons.
The triangle is just an indicator for the expansion, you should be able to just click somewhere to close the expanded view, but it could be a button as well.

![channel_profile_click](https://user-images.githubusercontent.com/61703250/77208857-0fa58f80-6afd-11ea-91bf-1164f366999d.png)

What do you think about this concept?

Credits:
- I used some icons from the default theme by [xpoke](https://github.com/xpoke/MetroMumble).
- I used some avatars by ryumo [deviantart](https://www.deviantart.com/ryumo/art/WoW-Avatar-Set-02-190098421), as profile examples.

**Comment by ImGunter** [comment]
This is a super neat concept tbh!

**Comment by Krzmbrzl** [comment]
@toby63 First off: thanks for the detailed explanation and illustration :+1:

I do think this would be a viable design choice to have. That being said though, _personally_ I don't like the idea of separating the server and the channel stuff. On the other hand the server-tab is pretty much as things work right now (and which I prefer) so maybe I wouldn't even be bothered by optionally being able to switch to a channel tab.

Regarding this issue as a whole:
From what I have seen in the code-base I have the impression that in order to mess with the UI and make significant changes to it (whatever that'll be exactly), we'd come at risk of breaking (core) functionality as the UI and the functionality is pretty intertwined at some places.
Of course that's definitely not a good design choice and I'm sure it hasn't been intended from the beginning. However a growing code base tends to grow in a way that is most convenient and not in a way that would be best for refactoring of that scale.
Thus I believe that before we can start turning the UI upside down, we have to cleanly separate the back- and frontend.

@davidebeatrici and I are already thinking about tackling that problem though so there are chances that this will be done in the (somewhat) near future. At this point I won't state any details or promises on this though.

**Comment by toby63** [comment]
To outline and summarize some general problems I see, that i would like to tackle with a new theme (combined with changes elsewhere):
- Chat: 
   - Is more important than you maybe think
   - Too much information goes into the chat, so we should seperate Channel-, Private- and Serverchat, as well as the log and notifications:
       - server chat is not necessary imo, so should maybe be optional (if active: activated by click on main channel maybe, so main chat window switches to server chat)
       - channel chat should be main chat (that is visible by default)
       - private chat should be seperated into own windows (with tabs for each conversation partner); see also #2521
       - log should only be accessable by menu (for example in "help" or "server")
       - notifications (like "user joined"):
                 - could be displayed in pop-ups (with option to disable) (con: might be annoying)  
                 - or in seperate "chat"-like row (for those who like to see it by default)
                 - or optional in normal chat window (so that I can disable it :stuck_out_tongue: )
                 - or disabled at all (sound notifications should be seperate; notifications in log)

   - The chat feels "hard" to use and read:
        - different position of chat (more central position) 
        - increased sizes, fonts
        - change input mask (more central position (not bottom), bigger)  
        - messages should be more distinguishable (I used boxes and colours); see also: #3538
        - add options to search and filter the text (like: only show messages from user)
                  - add hashtags to search for? (con: maybe to much twitter :smile:)

- Serverview:
   - Is a bit to simple imo:
        - Add symbols to channels; e.g. [xpaws theme](https://github.com/xPaw/FlatMumble) added the satelite symbol to make the channels more distinguishable from the users
        - Add options to add more information to channels (that is visible by default and/or visible in seperate row like in my concept)
        - more information and icons for users (e.g. status (like afk), profile pictures etc.)
   - Better visibility:
        - bigger fonts
        - add more space around the channels and users (maybe boxes around channels like I did)

**Comment by toby63** [comment]
@Krzmbrzl:
> That being said though, personally I don't like the idea of separating the server and the channel stuff. On the other hand the server-tab is pretty much as things work right now (and which I prefer) so maybe I wouldn't even be bothered by optionally being able to switch to a channel tab.

That was exactly my intention to keep the Server Tab fully functional on its own.
An additional idea would be to let people disable the Channel Tab in the options, if they feel bothered by it.

>we'd come at risk of breaking (core) functionality as the UI and the functionality is pretty intertwined
at some places. [...] Thus I believe that before we can start turning the UI upside down, we have to cleanly separate the back- and frontend.
>@davidebeatrici and I are already thinking about tackling that problem though so there are chances that this will be done in the (somewhat) near future. At this point I won't state any details or promises on this though.

That is unfortunate and promising at the same time.
I thought it could be easier to add a different UI, even though this is a long-term goal of course.

**Comment by toby63** [comment]
@ImGunter Why closing? The status as dicarded was applied automatically, so it is no decision.
I think this Issue can still be useful, if somebody wants to work on it in the future :).

**Comment by seniorm0ment** [comment]
If something like this is ever implemented, I'd prefer it just be a user theme. I really like Mumble's current minimal GUI, and hate Discord's style. However I understand people switching from Discord may want a similar style, so I think making Mumble fairly themeable would be the way to go.

**Comment by gitgrub** [comment]
Yes, please keep the ui simple. The quality of mumble is that its fast and lean.

**Comment by loicdelez** [comment]
I think the idea is very good and I was thinking to improve the concept, what could be good is to make a version by merging Mumble and Rocket.chat for the voice part and the chat part all in a nice design.

**Comment by gianmarcogg03** [comment]
Hello, I saw this issue today and decided to try making a simple mockup of the current Mumble UI but with tabs so that you can be connected to multiple servers at once, similarly to TeamSpeak 3, in case it might be useful. It's accurate with how tabs are done in Qt with the Breeze theme (the blue colors are from my KDE color scheme).

![Image](https://github.com/user-attachments/assets/d7473962-8958-4d64-b6f3-1332b3957fcb)
